### DISCUSSION TITLE
Converting casadi.mx data to numeric values

### DISCUSSION THREAD

**Discussion by sal-optimal** [casadi/casadi#3761] [Q&A]
Hello,

I haven't used casadi before, however, I came across the QuITO package for Matlab that solves optimal control problems. The package requires the installation of casadi first. I wrote a Matlab code to solve an optimal control problem and I got the solution (along with the graphs).  When I type "solution" which is a variable of type struct at the command prompt this is what I get:

*************************************
>> solution
solution = 
  struct with fields:

    time_taken: 13.6420
        output: [1×1 casadi.OptiSol]
             X: [1×101 casadi.MX]
         U_app: [1×101 casadi.MX]
           tau: [0 0.0500 0.1000 0.1500 … ]
          cost: 83.6700
*************************************

I would like to extract the data in the variables X and U_app which are of size 1 x 101 but I'm not sure how.
It is mentioned that X and U_app are casadi.MX variables, i.e., symbolic. How can I obtain the data for the state X and control signal U as numbers so that I make further calculations? I tried different ways includeing using the Matlab command struct2cell but with no luck.

Any help is appreciated.

Sal

**Comment by snDevX** [comment]
try this:
https://groups.google.com/g/casadi-users/c/osxRYjz-w64
